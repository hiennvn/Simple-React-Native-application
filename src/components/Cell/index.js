import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'

class Cell extends Component {
  handlePress = () => {
    this.props.onPress(this.props.id)
  }

  render() {
    return <View>{this.props.title}</View>
  }
}

Cell.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
}

export default Cell
