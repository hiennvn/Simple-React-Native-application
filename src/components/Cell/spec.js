import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import Cell from '.'

describe('Reusable Cell', () => {
  it('renders correctly', () => {
    expect(
      shallow(<Cell id="cell" title="Cell" onPress={jest.fn()} />)
    ).toMatchSnapshot()
  })
})
