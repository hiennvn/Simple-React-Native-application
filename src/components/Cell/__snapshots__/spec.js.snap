// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reusable Cell renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cell
    id="cell"
    onPress={[Function]}
    title="Cell"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": "Cell",
    },
    "ref": null,
    "rendered": "Cell",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": "Cell",
      },
      "ref": null,
      "rendered": "Cell",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`cell.js.snap 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cell
    id="cell"
    onPress={[Function]}
    title="Cell"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": "Cell",
    },
    "ref": null,
    "rendered": "Cell",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": "Cell",
      },
      "ref": null,
      "rendered": "Cell",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
